src/UnitTests.Shared/RunnerUtilities.cs

```diff
@@ -114,19 +113,23 @@ public static string RunProcessAndGetOutput(string process, string parameters, o
                 p.BeginOutputReadLine();
                 p.BeginErrorReadLine();
                 p.StandardInput.Dispose();
-                p.WaitForExit();
+                p.WaitForExit(30000);
 
+                pid = p.Id;
                 successfulExit = p.ExitCode == 0;
             }
 
             outputHelper?.WriteLine("==== OUTPUT ====");
             outputHelper?.WriteLine(output);
+            outputHelper?.WriteLine("Process ID is " + pid + "\r\n");
             outputHelper?.WriteLine("==============");
 
             Console.WriteLine("==== OUTPUT ====");
             Console.WriteLine(output);
+            Console.WriteLine("Process ID is " + pid + "\r\n");
```

Comment:
I'm likely not understanding the test, but process IDs get reused, and the algorithm used is not stable, so if a server process exits the next one can potentially get the same id?

Reply:
The tests should check the logic of msbuild server. That is:
- If the special flag is on, we use msbuild server for task execution. Msbuild server process is a long living one. It lives the same time as the msbuild nodes (15 min).
- If server is occupied, there should be a fallback to the usual, non-server, build.
The task itself writes down the id of the process that is used for the build, and here we write down the process id of the msbuild process that was created for building a project. If msbuild server was used, these ids would be different (first one would be the id of msbuild server and the second one would be the id of msbuild client). If not used, they would be the same. Thus, we check how the build was executed, in or out the server.